File: Programming Basics/C++/Conditional Statements Advanced/FishingBoat/FishingBoat.cpp
#include <iostream>
using namespace std;

int main()
{
    int budget, fishermenCount;
    string season;
    cin >> budget >> season >> fishermenCount;

    int rentPrice = 0;
    if (season == "Spring") rentPrice = 3000;
    else if (season == "Summer" || season == "Autumn") rentPrice = 4200;
    else if (season == "Winter") rentPrice = 2600;
    
    double discountMultiplier;
    if (fishermenCount <= 6) discountMultiplier = 0.9;
    else if (fishermenCount <= 11) discountMultiplier = 0.85;
    else discountMultiplier = 0.75;

    double totalCosts = rentPrice * discountMultiplier;

    if (fishermenCount % 2 == 0 && season != "Autumn") totalCosts *= 0.95;

    cout.setf(ios::fixed);
    cout.precision(2);
    if (budget >= totalCosts) cout << "Yes! You have " << budget - totalCosts << " leva left." << endl;
    else cout << "Not enough money! You need " << totalCosts - budget << " leva." << endl;
}

File: Programming Basics/JavaScript/Conditional Statements Advanced/Exercises/fishing-boat.js
function solve(input) {
    const budget = Number(input[0]);
    const season = input[1];
    const fishermenCount = Number(input[2]);

    let rentPrice = 0;
    if (season === "Spring") rentPrice = 3000;
    else if (season === "Summer" || season === "Autumn") rentPrice = 4200;
    else if (season === "Winter") rentPrice = 2600;

    let discountMultiplier = 0;
    if (fishermenCount <= 6) discountMultiplier = 0.9;
    else if (fishermenCount <= 11) discountMultiplier = 0.85;
    else discountMultiplier = 0.75;

    let totalCosts = rentPrice * discountMultiplier;
    if (fishermenCount % 2 === 0 && season != "Autumn") totalCosts *= 0.95;

    if (budget >= totalCosts) console.log(`Yes! You have ${(budget - totalCosts).toFixed(2)} leva left.`);
    else console.log(`Not enough money! You need ${(totalCosts - budget).toFixed(2)} leva.`);
}
